### ISSUE TITLE
Downtime:  error Broken pipe

### ISSUE THREAD

**Issue by Azef1** [shinken-monitoring/mod-webui#537]
Hi, 

With the latest version of WebUI, i have a problem when i schedule a downtime from all services view.
I have a 404 not found on http://myshinken:7767/forms/downtime/add/myhost/myservice with a empty popu
![capture1](https://cloud.githubusercontent.com/assets/15362041/18120976/9d16dd8c-6f62-11e6-893e-0d2723191dd2.PNG) 

In my brokerd.log, i see : 

> [WebUI] External command push, exception: [Errno 32] Broken pipe

Thx

**Comment by mohierf** [comment]
I do not think that the broken pipe error is because of the WebUI ... this error is raised when the external command can not be sent to the scheduler :(

Can you confirm that you do not have the problem with the previous version of WebUI with the same configuration?

**Comment by Azef1** [comment]
So, with same configuration:
With WebUI 2.3.2 --> No problem
With WebUI 2.4.2.c --> Schedule a downtime **from hosts or service**, works but Schedule a downtime **from http://myshinken:7767/all?search=type:service** not works i have a 404 not found with "Generic modal" empty.

**Comment by mohierf** [comment]
Ok. Thanks for reporting. I will have a look to this ...

**Comment by Azef1** [comment]
With Firebug i noticed a problem:
My services name contains space (eg; MySQL - TCP)
When i schedule a downtime, URL is `http://myShinken:7767/forms/downtime/add/myhost/MySQL`

I think the URL should be `http://myShinken:7767/forms/downtime/add/myhost/MySQL - TCP`

**Comment by mohierf** [comment]
Ok. But I think the URL should be: `http://myShinken:7767/forms/downtime/add/myhost/MySQL%20-%20TCP` ...

You should try to modify the file `htdocs/shinken-actions.js` on line 440 with:

```
   var url = '/action/SCHEDULE_'+elts.type+'_DOWNTIME/'+encodeURIComponent(elts.nameslash)+'/'+start_time+'/'+end_time+'/'+shinken_downtime_fixed+'/'+shinken_downtime_trigger+'/'+shinken_downtime_duration+'/'+user+'/'+comment;
```

Please confirm that it is ok with this modification ... I cannot make the test currently :wink:

**Comment by Azef1** [comment]
I did your modification and it isn't solved. I continue to find this bug...

**Comment by mohierf** [comment]
Ok. My bad ... I did not proposed the right modification ... sorry.

Replace line 194 of `footer_element.tpl` with: 

```
display_modal("/forms/downtime/add/"+encodeURIComponent(elt));
```

**Comment by Azef1** [comment]
Not exactly: 
Now URL generated: `http://myShinken:7767/forms/downtime/add/myHost%2FMySQL%20-%20TCP`
The problem is %2F, because if i replace %2F by / in my browser, the query works

**Comment by mohierf** [comment]
Ok, it is a little bit more complicated than I thought ... you should try to replace line 194 with:

```
            elt = elt.replace(' ', '%20');
            display_modal("/forms/downtime/add/"+elt);
```

**Comment by virtual-expo** [comment]
Hello,

I am facing a similar issue. I cannot add a downtime or a comment if my service name contains square brackets. For example:
http://myShinken/service/myHost/Very%20:%20[Long/Name]%20Check%20Something#downtimes
results in an empty "Generic Modal" popup.

I am also using WebUI 2.4.2.c.

**Comment by maethor** [comment]
Is it still an issue in the last version of the webui?
